postpic/particles/particles.py

```diff
@@ -62,11 +62,12 @@ class _SingleSpecies(object):
     used by the MultiSpecies class only.
     The _SingleSpecies will return atomic particle properties
     (see list below) as given by the dumpreader. Each property can thus
-    return
+    return:
     1) a list (one value for each particle)
-    2) a siingle scalar value if this property is equal for the entire
+    2) a single scalar value if this property is equal for the entire
     species (as usual for 'mass' or 'charge').
     3) raise a KeyError on request if the property wasnt dumped.
+    This class is immutable.
```

Comment:
Not sure if these changes really make the object really "immutable". The immutability here is really more a user guideline than a technical property, like with the field object. `Field.matrix[0] = 5` will also change a Field.
Maybe just remove this line?
A technically immutable object would need to use `__slots__` and read-only properties for any public member.